src/panels/lovelace/cards/hui-button-card.js

```diff
@@ -0,0 +1,132 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../../../components/ha-card.js';
+
+import { STATES_OFF } from '../../../common/const.js';
+import toggleEntity from '../common/entity/toggle-entity.js';
+import stateIcon from '../../../common/entity/state_icon.js';
+import EventsMixin from '../../../mixins/events-mixin.js';
+
+/*
+ * @appliesMixin EventsMixin
+ */
+class HuiButtonCard extends EventsMixin(PolymerElement) {
+  static get template() {
+    return html`
+    <style>
+      ha-icon {
+        display: flex;
+        margin: auto;
+      }
+      paper-button {
+        display: flex;
+        margin: auto;
+        text-align: center;
+      }
+      .not-found {
+        flex: 1;
+        background-color: yellow;
+        padding: 8px;
+      }
+    </style>
+    <ha-card on-click='_handleClick'>
+      <paper-button>
+        <div id='Button' class$='[[_computeClassName(_stateObj)]]'>
+          <ha-icon id='ButtonIcon' style='color: [[_computeColor(_stateObj)]]; width:40%; height:40%;"' icon='[[_computeIcon(_stateObj)]]'></ha-icon>
+          <span id='ButtonName'>[[_computeName(_stateObj)]]</span>
+          <span id='ButtonState'>[[_computeState(_stateObj)]]</span>
+        </div>
+      </paper-button>
+    </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object
+      },
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)'
+      },
+    };
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    if (!config || (!config.entity && !(config.action && config.service)) || (config.tap_action === 'call-service' && !config.service)) throw new Error('Invalid card configuration');
+    this._config = Object.assign({ color: 'var(--primary-text-color)', show_state: true }, config);
+    if (this._config.action) this._config.tap_action = 'call-service';
```

Comment:
What is the purpose of a separate `action` config? Can't it be handled with just `tap_action`

Reply:
Can't you just fail validation if `!config.entity && config.tap_action !== 'call-service'`. That will make entity required unless the button is for calling a service. `action` isn't intuitive

Reply:
I am open to suggestions. After you made that comment, I thought of a different way for the user to only call a service. Instead of the Action variable, if entity was null, then tap_action had to be call-service. if so then do the same as what action does. Thoughts?

If entity was null and tap_action not call-service. It would throw the invalid config error

Reply:
Okay, @jeradM I promise I did not see your comment before my last comment. but yea I think thats the best. Making changes

Reply:
I think that's what I was suggesting 😜. You don't ever need to test it again though. Just check in `setConfig` and throw if it fails